### PR TITLE
ASoC: SOF: Intel: hda-mlink: fix sublink refcounting

### PR DESCRIPTION
**Pull request by plbossart**
In hindsight it was a very bad idea to use the same refcount for Extended and 'legacy' HDaudio multi-links. The existing solution only powers-up the first sublink, which causes SoundWire and SSP tests to fail when more than one DAI is used concurrently. Solving this problem requires per-sublink refcounting, as suggested in this patch.

The existing refcounting remains for 'legacy' HdAudio links, mainly to avoid changing the obscure programming sequence in snd_hdac_ext_bus_link_put().

### CODE REVIEW

**Review by ranj063 — APPROVED** [review]

**Comment by greg-intel** [comment]
During testing, sh-glk-bob-da7219-03 failed, it looks to have run out of space, investigating further.

**Comment by greg-intel** [comment]
SOFCI TEST

**Comment by keqiaozhang** [comment]
SOFCI TEST

**Comment by plbossart** [comment]
@ujfalusi @bardliao can you please take a look, thanks!

File: sound/soc/sof/intel/hda-mlink.c [hunk 1]
```diff
@@ -641,8 +646,13 @@ static int hdac_bus_eml_power_up_base(struct hdac_bus *bus, bool alt, int elid,
 	if (eml_lock)
 		mutex_lock(&h2link->eml_lock);
 
-	if (++hlink->ref_count > 1)
-		goto skip_init;
+	if (!alt) {
```
Comment (bardliao) [inline sound/soc/sof/intel/hda-mlink.c:649]: @plbossart  What does `alt1` stand for? Is `!alt` for legacy' HdAudio links?

Comment (plbossart) [reply sound/soc/sof/intel/hda-mlink.c:649]: yes, it's the definition in the HAS

LCAP::ALT: BIT(28):
Indicates which Link Type this link belongs to.
0000 = Intel HD Audio Link
0001 = Extended Audio Link

see also https://github.com/thesofproject/linux/blob/topic/sof-dev/Documentation/sound/hd-audio/intel-multi-link.rst for a more visual representation

**Review by bardliao — APPROVED** [review]

**Pull request merged**